src/main/java/com/utilaider/logistics/controller/LogisticsController.java

```diff
@@ -178,9 +218,266 @@ public String getProfile(@ModelAttribute Vehicle vehicle) {
                 UserDetails userDetails = (UserDetails) auth.getPrincipal();
                 Owner owner = ownerService.getOwnerByUsername(userDetails.getUsername());
                 owner.getVehicles().add(vehicle);
-                ownerService.updateOwner(owner);
+                ownerService.update(owner);
+                return "redirect:home";
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+            return "redirect:home";
+        }
+    }
+    
+    @RequestMapping(value = "adddriver", method = RequestMethod.POST)
+    public String addDriver(@ModelAttribute Driver driver, HttpServletRequest request) {
+        try {
+            Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+            if ((auth instanceof AnonymousAuthenticationToken)) {
+                return "redirect:login";
+            } else {
+                UserDetails userDetails = (UserDetails) auth.getPrincipal();
+                Owner owner = ownerService.getOwnerByUsername(userDetails.getUsername());
+                driver.setIsActive(Boolean.TRUE);
+                driver.setIsDeleted(Boolean.FALSE);
+                driver.setPassword(new BCryptPasswordEncoder().encode((driver.getFirstName() + "123")));
```

Comment:
Currently We have no feature to driver add own-self in system.
So I had use first-name with 123.